### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by NATASHA-ct**
### Project requirements ✔️ 
- Install CanCanCan in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -14,4 +14,10 @@ class User < ApplicationRecord
   def three_most_recent_posts
     posts.last(3)
   end
+
+  %i[admin default].freeze
+
+  def admin?(requested_role)
+    role == requested_role.to_s
+  end
```
Comment (BenMukebo) [inline app/models/user.rb:22]: - Tokos, well done 😃 👍🏻

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -30,4 +30,13 @@ def create
       end
     end
   end
+
+  def destroy
+    @post = Post.find(params[:id])
```
Comment (BenMukebo) [inline app/controllers/posts_controller.rb:36]: - [ ] Great job adding authentication from the view 👏🏻 But you need to add this `load_and_authorize_resource` as weel to the top of your Post Controller and the  Comment controller to make sure your authorization rules are enforceable and use **CanCanCan** for post and comment authorization 😉 👍🏻 

- Note: you could also add `authorize_resource` if you had this line eg;
`Before_action:set_name, only: %[show update destroy] ` at the top of your controller

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @NATASHA-ct, @karlavdelgadof,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Has CanCanCan `ability` class that defines the set of rules for user - post interactions.✔️ 
- `role` column exist in the users table.✔️ 
- Professional README ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Implement the required change👏 👏
-  Add a role-based authorization to the App using cancancan 👏 👏
- Linter is green (the last commit has a green check mark (✅)) 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**